pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/OfflineClusterIntegrationTest.java

```diff
@@ -124,6 +165,15 @@ public void setUp() throws Exception {
     }
   }
 
+  @Test
+  public void testInstancesStarted() {
```

Comment:
This test did not catch the infinite loop inside getServiceStatus. Better to expose this callback into a rest call, and call server/broker to check the status on their side?